src/server/server_family.cc

```diff
@@ -502,6 +506,25 @@ void ServerFamily::Init(util::AcceptServer* acceptor, std::vector<facade::Listen
     return res.has_value();
   });
 
+  // We only reconfigure TLS when the 'tls' config key changes. Therefore to
+  // update TLS certs, first update tls_cert_file, then set 'tls true'.
+  config_registry.RegisterMutable("tls", [this](const absl::CommandLineFlag& flag) {
+    if (!ValidateServerTlsFlags()) {
```

Comment:
Though if you start Dragonfly without TLS or a password, then enable TLS, you should be blocked as you haven't set a password? So in `Serverfamily::ServerFamily` TLS would be false so does nothing?

Reply:
You call `ValidateServerTls` in `ServerFamily` constructor and you exit if:

```
  if (!ValidateServerTlsFlags()) {
    exit(1);
  }
```

So, the basically, we exit if there is no auth set (either via password or ca_dir/cert).

> Though if you start Dragonfly without TLS or a password

Is not true because at least one of them must be set, the semantics of that function is roughly "At some auth must be set`. That way, if it succeeds and we don't exit, then it can never be the case that both `password` and  `ca_cert/ca_dir` are empty.

Reply:
> So, the gist is that we exit if there is no auth set (either via password or ca_dir/cert).

Only if TLS is enabled though - you can start Dragonfly with neither set - then enable TLS later

> Is not true because at least one of them must be set, the semantics of that function is roughly

I'm still not sure I understand - as an example - removing the `ValidateServerTlsFlags` check:
```
./dragonfly --alsologtostderr &
redis-cli config set tls_cert_file certs/df-cert.pem
redis-cli config set tls_key_file certs/df-key.pem
redis-cli config set tls true
redis-cli --tls --insecure ping
PONG
```

So that doesn't have either client auth or a password?

Whereas with that check:
```
redis-cli config set tls true
(error) ERR CONFIG SET failed (possibly related to argument 'tls') - argument can not be set
```
And Dragonfly logs `E20231023 16:14:49.619365 3338585 server_family.cc:318] TLS configured but no authentication method is used!`

Reply:
> Only if TLS is enabled though - you can start Dragonfly with neither set - then enable TLS later

This answers it. You are right, I did not think of this case :)

Reply:
> I think you can safely remove this. The reason is that the only way `ValidateServerTls` can fail, is if `hasAuth` is false which we `exit(1)` if it's not set in `ServerFamily::ServerFamily`. Otherwise, either the `password` or the `tls flags` are set.
> 
> Therefore, we only care if `requirepass` or `FLAGS_tls_ca_cert_file` are both empty at the same time. But this can't happen, because they can't be set to empty via `CONFIG SET` (or at least I don't see how they can be set to empty because `CONFIG SET requirepass` won't work for exmaple)

`config set dir ""` works